code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessorTests.kt

```diff
@@ -333,8 +349,21 @@ class AnalyticsHitProcessorTests {
 
     @Test
     fun `network success - assurance enabled`() {
+        // Set up mock Assurance Shared State
+        val mockAssuranceSharedStateData = mapOf("integrationid" to "testIntegrationId")
```

Comment:
Add more tests for negative cases:

- No integrationid in sharedstate
- Null integrationid in sharedstate
- Empty String integrationid in sharedstate

Reply:
Also the case showing
will not be able to extract the integration id when sharedState status is `Pending`